opentelemetry-sdk/tests/logs/test_handler.py

```diff
@@ -86,105 +88,114 @@ def test_log_flush_noop(self):
 
-    def test_log_record_no_span_context(self):
-        processor, logger = set_up_test_logging(logging.WARNING)
+        # teardown
+        logger.removeHandler(handler)
 
-        # Assert emit gets called for warning message
-        with self.assertLogs(level=logging.WARNING):
-            logger.warning("Warning message")
+    def test_log_record_no_span_context(self):
+        with set_up_test_logging(logging.WARNING) as (processor, logger):
+            # Assert emit gets called for warning message
+            with self.assertLogs(level=logging.WARNING):
+                logger.warning("Warning message")
 
-        log_record = processor.get_log_record(0)
+            log_record = processor.get_log_record(0)
 
-        self.assertIsNotNone(log_record)
-        self.assertEqual(log_record.trace_id, INVALID_SPAN_CONTEXT.trace_id)
-        self.assertEqual(log_record.span_id, INVALID_SPAN_CONTEXT.span_id)
-        self.assertEqual(
-            log_record.trace_flags, INVALID_SPAN_CONTEXT.trace_flags
-        )
+            self.assertIsNotNone(log_record)
+            self.assertEqual(
+                log_record.trace_id, INVALID_SPAN_CONTEXT.trace_id
+            )
+            self.assertEqual(log_record.span_id, INVALID_SPAN_CONTEXT.span_id)
+            self.assertEqual(
+                log_record.trace_flags, INVALID_SPAN_CONTEXT.trace_flags
+            )
 
     def test_log_record_observed_timestamp(self):
-        processor, logger = set_up_test_logging(logging.WARNING)
-
-        with self.assertLogs(level=logging.WARNING):
-            logger.warning("Warning message")
+        with set_up_test_logging(logging.WARNING) as (processor, logger):
+            with self.assertLogs(level=logging.WARNING):
+                logger.warning("Warning message")
 
-        log_record = processor.get_log_record(0)
-        self.assertIsNotNone(log_record.observed_timestamp)
+            log_record = processor.get_log_record(0)
+            self.assertIsNotNone(log_record.observed_timestamp)
 
     def test_log_record_user_attributes(self):
         """Attributes can be injected into logs by adding them to the LogRecord"""
-        processor, logger = set_up_test_logging(logging.WARNING)
-
-        # Assert emit gets called for warning message
-        with self.assertLogs(level=logging.WARNING):
-            logger.warning("Warning message", extra={"http.status_code": 200})
+        with set_up_test_logging(logging.WARNING) as (processor, logger):
+            # Assert emit gets called for warning message
+            with self.assertLogs(level=logging.WARNING):
+                logger.warning(
+                    "Warning message", extra={"http.status_code": 200}
+                )
 
-        log_record = processor.get_log_record(0)
+            log_record = processor.get_log_record(0)
 
-        self.assertIsNotNone(log_record)
-        self.assertEqual(len(log_record.attributes), 4)
```

Comment:
Not sure why this was removed. Attributes should stay the same even when formatted.